core/src/test/java/org/commonjava/maven/ext/manip/util/PropertyInterpolatorTest.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.maven.artifact.repository.DefaultArtifactRepository;
+import org.apache.maven.artifact.repository.layout.DefaultRepositoryLayout;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.settings.Settings;
+import org.commonjava.maven.atlas.ident.ref.SimpleProjectVersionRef;
+import org.commonjava.maven.ext.manip.io.ModelIO;
+import org.commonjava.maven.ext.manip.model.Project;
+import org.commonjava.maven.ext.manip.resolver.GalleyAPIWrapper;
+import org.commonjava.maven.ext.manip.resolver.GalleyInfrastructure;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class PropertyInterpolatorTest
+{
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    @Test
+    public void testInteropolateProperties() throws Exception
+    {
+        final Model model = resolveRemoteModel( "org.infinispan:infinispan-bom:8.2.0.Final" );
+
+        Project p = new Project( model );
+
+        Properties props = p.getModel().getProperties();
+        boolean containsProperty = false;
+        for ( Object o : props.values() )
+        {
+            if ( ( (String) o ).contains( "${" ) )
+            {
+                containsProperty = true;
+            }
+        }
+        assertTrue( containsProperty );
+        PropertyInterpolator pi = new PropertyInterpolator( props, p );
+        assertTrue( pi.interp( "${version.hibernate.osgi}" ).equals( "5.0.4.Final" ) );
+    }
+
+    @Test
+    public void testInteropolateDependencies() throws Exception
+    {
+        final Model model = resolveRemoteModel( "org.infinispan:infinispan-bom:8.2.0.Final" );
+        Project project = new Project( model );
+        PropertyInterpolator pi = new PropertyInterpolator( project.getModel().getProperties(), project );
+
+        String nonInterp ="" , interp = "";
+        for ( Dependency d : project.getManagedDependencies())
+        {
+            nonInterp += ( d.getGroupId().equals( "${project.groupId}" ) ? project.getGroupId() : d.getGroupId() ) + ":" +
+                            ( d.getArtifactId().equals( "${project.artifactId}" ) ? project.getArtifactId() : d.getArtifactId() ) + System.lineSeparator();
+
+            interp += pi.interp( d.getGroupId().equals( "${project.groupId}" ) ? project.getGroupId() : d.getGroupId() ) + ":" +
+                                pi.interp( d.getArtifactId().equals( "${project.artifactId}" ) ? project.getArtifactId() : d.getArtifactId() ) + System.lineSeparator();
+
+        }
+        assertTrue ( nonInterp.contains( "${" ) );
+        assertFalse ( interp.contains( "${" ) );
+    }
+
+    private Model resolveRemoteModel( final String resourceName ) throws Exception
+    {
```

Comment:
Looks like this is used a few times...is it worth making a utils class (stateless) or a test fixture (stateful)?